lib/src/data/datasource/remote/pokedex_api_service.dart

```diff
@@ -0,0 +1,55 @@
+import 'dart:convert';
+import 'dart:io';
+import '../../../core/util/constants.dart';
+import '../../../core/resource/data_state.dart';
+import '../../../domain/entity/pokedex_entity.dart';
+import '../../model/pokedex_model.dart';
+import 'package:http/http.dart' as http;
+import '../../model/pokemon_model.dart';
+import '../../model/sprites_model.dart';
+
+class PokedexApiService {
+  final client = http.Client();
+
+  Future<DataState<PokedexEntity>> getPokemonIds() async {
+    try {
+      final response = await client.get(Uri.parse(Constants.pokedexUrl));
+      if (response.statusCode == HttpStatus.ok) {
+        final pokedexModel = PokedexModel.fromJson(
+          json.decode(response.body),
+        );
+        for (int i = Constants.initialValue;
+            i < pokedexModel.pokemonEntries.length;
+            i++) {
+          pokedexModel.pokemonEntries[i].pokemon =
+              await _getPokemon(pokedexModel.pokemonEntries[i].entryNumber);
+        }
```

Comment:
Instead to this, you can use a forEach iteration

Reply:
I tried but the await does not work in the foreach

Reply:
you can try using Future.forEach in this case

Reply:
you can try Future.forEach in this case